ako-infra/avirest/handle_netinfo.go

```diff
@@ -0,0 +1,250 @@
+	lslrmap := lib.GetNetinfoCRData()
+	utils.AviLog.Infof("Got data LS LR Map: %v, from NetworkInfo CR", lslrmap)
+
+	client := avicache.SharedAVIClients().AviClient[0]
+	found, cloudModel := getAviCloudFromCache(client, utils.CloudName)
+	if !found {
+		utils.AviLog.Warnf("Failed to get Cloud data from cache")
+		return
+	}
+	matched := matchSegmentInCloud(cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs, lslrmap)
+	if !matched {
+		cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs = constructLsLrInCloud(lslrmap)
+		path := "/api/cloud/" + *cloudModel.UUID
+		restOp := utils.RestOp{ObjName: utils.CloudName, Path: path, Method: utils.RestPut, Obj: &cloudModel,
+			Tenant: "admin", Model: "cloud", Version: utils.CtrlVersion}
+		executeRestOp("fullsync", client, &restOp)
+	} else {
+		utils.AviLog.Infof("LS LR update not required in cloud: %s", utils.CloudName)
+	}
+}
+
+func AddSegment(obj interface{}) {
+	objKey := "Netinfo" + utils.ObjKey(obj)
+	utils.AviLog.Debugf("key: %s, Network Info DELETE Event", objKey)
+	crd := obj.(*unstructured.Unstructured)
+
+	specJSON, found, err := unstructured.NestedStringMap(crd.UnstructuredContent(), "topology")
+	if err != nil || !found {
+		utils.AviLog.Warnf("NetworkInfo topology not found: %+v", err)
+		return
+	}
+	lr := specJSON["gatewayPath"]
+	ls := specJSON["aviSegmentPath"]
+	utils.AviLog.Infof("key: %s, Adding LR %s, LS %s from networkinfo CR", objKey, lr, ls)
+	client := avicache.SharedAVIClients().AviClient[0]
+	found, cloudModel := getAviCloudFromCache(client, utils.CloudName)
+	if !found {
+		utils.AviLog.Warnf("key: %s, Failed to get Cloud data from cache", objKey)
+		return
+	}
+	updateRequired, lslrList := addSegmentInCloud(cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs, lr, ls)
+	if updateRequired {
+		cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs = lslrList
+		path := "/api/cloud/" + *cloudModel.UUID
+		restOp := utils.RestOp{ObjName: utils.CloudName, Path: path, Method: utils.RestPut, Obj: &cloudModel,
+			Tenant: "admin", Model: "cloud", Version: utils.CtrlVersion}
+		executeRestOp(objKey, client, &restOp)
+	} else {
+		utils.AviLog.Infof("key: %s, LSLR update not required in cloud: %s", objKey, utils.CloudName)
+	}
+}
+
+func DeleteSegment(obj interface{}) {
+	objKey := "Netinfo" + utils.ObjKey(obj)
+	utils.AviLog.Debugf("key:%s, Network Info DELETE Event", objKey)
+	crd := obj.(*unstructured.Unstructured)
+
+	specJSON, found, err := unstructured.NestedStringMap(crd.UnstructuredContent(), "topology")
+	if err != nil || !found {
+		utils.AviLog.Warnf("key: %s, NetworkInfo topology not found: %v", objKey, err)
+		return
+	}
+	lr := specJSON["gatewayPath"]
+	ls := specJSON["aviSegmentPath"]
+	utils.AviLog.Infof("key: %s, Network Info CR deleted, removing LR %s, LS %s from cloud", objKey, lr, ls)
+	client := avicache.SharedAVIClients().AviClient[0]
```

Comment:
It seems when the WCP cluster is being teared down, we will see lot of these `DeleteSegment` calls and because we are using the same AviClient, all calls would be blocked into 1 session. Can we do something to make these parallel, without necessary using a queue?